pycaret/tests/conftest.py

```diff
@@ -56,6 +56,7 @@ def load_ts_models(load_setup):
         "gpu_param": False,
         "X_train": pd.DataFrame(get_data("airline")),
         "enforce_pi": False,
+        "sp_to_use": 1,
```

Comment:
By definition, `sp_to_use` will at least be 1, so I add it in this fixture.

Reply:
Will it not be 12 in this case? Since data is monthly seasonal (airline dataset)?

Reply:
12 definitely makes sense, because for the `airline` dataset, if we don't specify a `seasonal_period`, that's the seasonality that would be inferred from the data and the value that `sp_to_use` will end up having. But that would change the tests as they were before my change and I didn't wanted to introduce such change silently. 

Before my change, all models had:
https://github.com/pycaret/pycaret/blob/e153069683166a2096567e5e4b01df4132e4ad65/pycaret/containers/models/time_series.py#L513-L514
which means that `self.sp` ended up being 1 in the tests.

Other than that, I see no problem in changing the value. Let me know what you think.

Reply:
But `sp_to_use` would have been 12 before the change as well (not 1). So before the change, `self.sp` (in the model container) would end up taking a value of 12 also (not 1). Did I miss something here?

![image](https://user-images.githubusercontent.com/33585645/151699080-da79d452-26e7-4408-b5e0-1cdc225b4278.png)

Reply:
You can confirm this by creating an ARIMA model (using `time_series` branch which does not have your changes yet). You will see that it will take sp = 12 in the model.

Reply:
Yes, this one is a bit tricky. I think we are both right but talking about different things.
* Yes, following the regular execution path, if you don't specify `seasonal_period` in the setup, `sp_to_use` is inferred from the data, and ends up being 12.
* However, if we take a look at one of the tests that fail if I don't add this `sp_to_use=1`:
https://github.com/pycaret/pycaret/blob/b1502037b87d035b5d75585a31eb1ac0e9cdce7e/pycaret/tests/test_time_series_blending.py#L43
it uses the fixture `load_models`, which in turn calls `get_all_model_containers`, triggering the init method of all time series models, and it does so with a `globals_dict` that cannot really happen in a regular code execution path: it has no `sp_to_use`.  
https://github.com/pycaret/pycaret/blob/b1502037b87d035b5d75585a31eb1ac0e9cdce7e/pycaret/tests/conftest.py#L78-L86
Thus, before my changes, `self.sp` ends up being 1 in the models initialization:
https://github.com/pycaret/pycaret/blob/b1502037b87d035b5d75585a31eb1ac0e9cdce7e/pycaret/containers/models/time_series.py#L399-L400
After my changes, however, models initialization don't accept a `self.sp=None` and fail if I don't specify an `sp_to_use` in `load_models``:
https://github.com/pycaret/pycaret/blob/e153069683166a2096567e5e4b01df4132e4ad65/pycaret/containers/models/time_series.py#L399
https://github.com/pycaret/pycaret/blob/e153069683166a2096567e5e4b01df4132e4ad65/pycaret/containers/models/time_series.py#L434

This is inconsequential really because in the `load_models` fixture, `load_setup` is called afterwards, so the regular setup method is called and models are finally created through `create_model`, having the right `sp_to_use` which will be 12, no matter the value specified in `load_models`.

https://github.com/pycaret/pycaret/blob/b1502037b87d035b5d75585a31eb1ac0e9cdce7e/pycaret/tests/conftest.py#L65-L72

So tl;dr, it really does not matter the value that we set up here, it is only used in the models init triggered by `get_all_model_containers` , and the call `to `get_all_model_containers` is only used to get all model ids. These ids are then used to create proper model instances which will have the right `sp_to_use=12` value. But what will make more sense probably is directly specifying here `sp_to_use=12`. I'm changing the value to 12.

Reply:
Thank you for the detailed analysis, clarifying, and changing it to 12. I think at a high level it sounds reasonable, but the only issue might be that when `sp_to_use = 1`, some models like `snaive` (seasonal naive) are disabled. But `sp_to_use = 12` (value > 1) might be safer for that reason.